vowpalwabbit/active.cc

```diff
@@ -171,11 +171,11 @@ base_learner* active_setup(options_i& options, vw& all)
   // Create new learner
   learner<active, example>* l;
   if (options.was_supplied("simulation"))
-    l = &init_learner(data, base, predict_or_learn_simulation<true>, predict_or_learn_simulation<false>);
+    l = &init_learner(data, base, predict_or_learn_simulation<true>, predict_or_learn_simulation<false>, "active-simulation", false);
```

Comment:
so how would you explain to someone making a new reduction whether or not they need to call into predict before learning? is it based on the reduction itself they are coding or on the stack or what else? or how are you planning on document this change in learner by adding "predict_before_learn" bool

Reply:
In new reductions, if learn does not need prediction, the default work.  This is the common expected use case and default arguments for init functions are setup this way.

If reductions need predictions during learn, then things need to considered more carefully for efficiency reasons.  If there is a lot of extra setup for predictions, and the same work needs to happen for learn then predict_before_learn is set to false (for example ccb).  In this case learn for a reduction must call predict during learn.